### ISSUE TITLE
Consider using FOSSRIT Fedora mailing list for community discussion

### ISSUE THREAD

**Issue by jwflory** [FOSSRIT/fossbox-tasks#30]
# Summary

We recently discovered a [FOSSRIT mailing list](https://lists.fedoraproject.org/admin/lists/fossrit.lists.fedorahosted.org/) hosted by the Fedora Project in IRC. It may benefit us to use this mailing list as a general communication method for students, alumni, and others who don't use IRC or Telegram but want to stay involved.

# Description

_Unlike the RIT-hosted FLOSS Seminar list_, the Fedora mailing list hosted there uses Mailman3 / Hyperkitty. What this means is that there is a nice, pretty GUI that anyone can use to interact and engage with the mailing list (and it still works with regular email fine). There's also multiple ways to sign in and subscribe, with Facebook, Twitter, GitHub, GitLab, and more sign-in options, so the accessibility of this list is much greater than the Mailman2 instance we're using for the FLOSS Seminar list.

Additionally, the FLOSS Seminar list, by its name, was intended to be used only for the HFOSS course. But since we don't have any other mailing list (other than the announce-only list managed by @itprofjacobs and @schneidy), I think using this list for general community news / updates is a good way to engage with more people who are still interested in FOSSRIT and want to participate more, but maybe don't use an IRC client or don't like Telegram (or just find it difficult to keep up with the activity there).

To help expedite the process, should we choose to do this, I've [requested admin privileges](https://pagure.io/fedora-infrastructure/issue/5682) of the mailing list.

# Implementation

1. Gain privileges to the list, add FOSS@MAGIC faculty / staff to admin list
2. Change any settings or customize as necessary (if needed at all)
3. Post announcement to the announce-only FOSS@RIT list and the FLOSS Seminar list
4. Add a link to the mailing list in the IRC channel topic
5. Edit the Telegram pinned message to include the mailing list link

???

**Comment by itprofjacobs** [comment]
Seems to make sense.  Would like to hear from DJoe and Dan on it

**Comment by Nolski** [comment]
I'm a fan of this. Perhaps if we had multiple admins which included students to help on board new people. What's the fedora regulations for handing out those sort of privileges?

**Comment by jflory7** [comment]
@Nolski For the FedoraHosted lists, it's all managed by us. I know there are three "roles" people can have: subscribers, moderators, and owners. Owners are basically full admins, and we'd only want to give that to a small number of people. Moderators can approve and reject messages in the queue (e.g. if someone posts to the list and forgot to subscribe first, or just rejecting spam), so I think it would be a good idea to have some active students as moderators too.

**Comment by ritjoe** [comment]
Not every group has enough traffic to warrant it, but I'm generally a fan of having a diversity of mailing lists, including an "announce" kind of list that's low-traffic for events and happenings, and a "discussion" kind of list. 

The RIT-hosted GNU Mailman v2 list that Dan manages seems to fill the role already of the low-traffic "announce" list. 

I wasn't very jealous about the RIT-hosted, revived floss-seminar list being kept just to class discussion, but the name seems to have inhibited the broader community from participating. If we try to keep that academic-focussed, that's fine by me.

The Fedora-hosted list seems to have been a predecessor to the floss-seminar list (or maybe ran in parallel? or just different semesters with different instructors?). Anyway, if we revive that and take it over to fill the "discussion" role for a broader community of alumni; students enrolled at RIT and involved in the FOSS community, but not necessarily enrolled in one of the FOSS courses; friends of RITFOSS; and so on, that would be fine by me. 

I haven't used the GNU Mailman v3 aka "hyperkitty" version that's running there, so seeing how that goes has its appeal, too.

I like the idea of recycling (after a fashion) these web resources, too, rather than leaving them cobwebbed and abandoned, or flushing them down the memory hole.

**Comment by jflory7** [comment]
Today, I received owner rights to the mailing list and I've gone ahead and made some configuration changes to the list. As of now, I can also add anyone else as an admin or moderator to the list as needed. But I think in order for me to do that, one has to subscribe first.

Also, I posted [this short announcement](https://lists.fedoraproject.org/archives/list/fossrit@lists.fedorahosted.org/thread/YAY3SZBZ25XN2X7BWRG4QN6F7HIXI6CA/) to let anyone hanging out on that list that we're talking about reviving it here in this ticket.

**Comment by schneidy** [comment]
We should compare who is on each list. From @jflory7's description of GNU Mailman v3, I would be very willing to give the other list a try. The fact that it is easier to subscribe to the mailing list is huge.

**Comment by jlew** [comment]
Hi,

I was one of the original founding members.  Just to add some background to the conversation, this list was for the greater vision of FOSS@RIT (pre-magic days) and wasn't tied directly to the course. 

In the early days, the original course that was a seminar devoted to the development of educational software for the one laptop per child XO Laptop.  During that course, a few of the students and staff got together and started the foss@rit group, a group for students to learn, mentor, and interact with open source communities.  As the FOSS@RIT group picked up traction, the class evolved into "Seminar in Humanitarian Open Source Development".  Around that time RIT introduced its Minor in FOSS and created MAGIC.

The mailing list never had a lot of traffic as we had quite a few lists that were specific to different projects (class and outside of class).  The main method of communication was the IRC channels and the hope was the fossrit would become a place for people who didn't lurk on IRC to get periodic updates on the happenings.

**Comment by jflory7** [comment]
@jlew Thanks for the background info here! I'd like to go ahead and suggest that we begin promoting the mailing list fully across our IRC channel (in the topic) and also put out an announcement to the FossMagic announce list on lists.rit.edu.

**Comment by jflory7** [comment]
So it seems like a general consensus on this idea. Do we want to go ahead and begin putting this mailing list on websites / IRC channel topic / other places to point people towards?

**Comment by jflory7** [comment]
@ritjoe Think you could include the link to this mailing list in the channel topic?

**Comment by jflory7** [comment]
Hey everyone, just wanted to follow up here. Seems like we all agreed that it would be helpful to start using it (and we have been for the past couple months).

Would we want to add it to the homepage on [foss.rit.edu](http://foss.rit.edu) and the `#rit-foss` channel topic? Or do we still have anything we want to follow up on here?

This is the link we can share and promote if we add it to the website / channel topic: https://lists.fedoraproject.org/admin/lists/fossrit.lists.fedorahosted.org/

**Comment by Nolski** [comment]
Sounds like a good idea to me. Created a pull request here.

https://github.com/FOSSRIT/fossrit.github.io/pull/8

**Comment by jflory7** [comment]
@Nolski Awesome, thanks! 💯 💯 💯 

@schneidy @itprofjacobs Do you think you could start the process to get this added to the MAGIC webpage somewhere?

@ritjoe We had briefly discussed this a while back, but do you think it would be possible to get this URL added to the channel topic?

**Comment by ritjoe** [comment]
So, the question has arisen as to nature of the fedorahosted list. 

I want to be certain that we are not adding addresses to the list. Instead, we should let people manage their own subscriptions, both for subscribing and then for unsubscribing.

Currently, we're seeing messages from people asking to be unsubscribed, eg

https://lists.fedoraproject.org/archives/list/fossrit@lists.fedorahosted.org/message/6WIEDSLAOEJ6QXIWDW4PQ5PRTHQYMQ75/

but also one that has been held in the moderation queue and didn't make it to the web archive.

It is reputationally vital that we do not add addresses for people that are expecting to get an announcements-only like volume of email, because they will be upset as we ratchet up the level of discussion on the list as planned in this thread earlier.

@schneidy we had a discussion about this a few weeks ago. If you have added addresses, could you please be sure to remove them?

(There is the added twist that I am receiving admin messages, but cannot act on them. It's a long story, perhaps better told later or elsewhere, but suffice for now to say we are working on it.)

The bottom line is, if we get a request that someone's address to be removed from the list, I'd rather start that discussion with "you should unsubscribe through the same mechanism that you subscribed".

In part this encourages people to take responsibility for these things themselves. Also, it can be a tricky business knowing what addresses they are using (it is typical at RIT for students, faculty, and staff to be using a complex web of forwardings and therefor to be a bit out to see as to how it works), and whether the request is legitimate or not. 

But, asking them to unsubscribe themselves is not a reasonable place to start the discussion if someone else subscribed them.

**Comment by ritjoe** [comment]
@schneidy 

Could you please respond with some information about whether, or not, you've added addresses of other people to this list, carried over from the RIT-hosted GNU Mailman 2 announce-only list?

At the very least, knowing whether this was done, or not, helps us respond appropriately to requests like:

https://lists.fedoraproject.org/archives/list/fossrit@lists.fedorahosted.org/message/DMHONCVQEPD5EFZVADHC5CQDNV6JIPXA/

If they subscribed themselves, it would be completely proper to point them to the page that allows them to unsubscribe themselves, since that's how their relationship with the list started in the first place (eg, for short: opt-in).

If their address was added by someone else, then it would be much more obnoxious to point them at that interface and expect them to navigate it. The appropriate response, in that case, would be to carry out fulfilment of their request ourselves.

**Comment by jflory7** [comment]
@ritjoe Given the previous discussions in person about this, do you think we can update the channel topic for the mailing list now?

**Comment by ritjoe** [comment]
Sorry, I should configure my filters to prioritize FOSSRIT stuff over the RITLUG github notifications, but I haven't, so I overlooked this.

Sure, I'll change the topic.

Given offline discussions @Schneidy and I had, I think we share a better understanding of the concerns I raised above in

https://github.com/FOSSRIT/fossbox-tasks/issues/30#issuecomment-299738460 and https://github.com/FOSSRIT/fossbox-tasks/issues/30#issuecomment-302077904

but my feeling is that we didn't come to a firm decision about how to use the various lists, but rather that we just ran out the clock on the discussion at the end of the Spring 2017 semester with conferences, Imagine, finals, graduation, and the seasonal student diaspora.

So, with the list being advertised I'll change the topic, but I think this issue should remain open until we get that sorted.

**Comment by ritjoe** [comment]
Done:

```
18:56 -!- mode/#rit-foss [+o dzho] by ChanServ
18:56 -!- Topic for #rit-foss: The best things in life are FOSS@RIT | Center of gravity for all things 
          Free/Open Source in RIT's academic FOSS programs. See http://fossrit.github.io & 
          https://github.com/FOSSRIT/fossrit.github.io/issues/4#issuecomment-272707046
18:56 -!- Topic set by foss_bot [~limnoria@saskia.igm.rit.edu] [Fri May 12 14:32:14 2017]
18:58 -!- dzho changed the topic of #rit-foss to: The best things in life are FOSS@RIT | Center of gravity 
          for all things Free Software & Open Source in RIT's academic FOSS programs. See 
          http://fossrit.github.io & 
          https://lists.fedoraproject.org/archives/list/fossrit@lists.fedorahosted.org/
```

**Comment by ritjoe** [comment]
On Tue, Nov 21, 2017 at 07:53:33PM +0000, Justin W. Flory wrote:
>     @ritjoe said…
>     but my feeling is that we didn't come to a firm decision about how to use
>     the various lists, but rather that we just ran out the clock on the
>     discussion at the end of the Spring 2017 semester with conferences,
>     Imagine, finals, graduation, and the seasonal student diaspora.
> 
> Could you clarify what needs to be decided on? I'm trying to get an idea of
> what's left to move this forward.

To be very brief and to avoid repeating myself: 

Some clear policy with regard to how we subscribe and unsubscribe addresses
to the FOSSRIT Fedora mailing list in particular.

Some clear understanding as to how the list is to be used.

**Comment by jflory7** [comment]
> _**@ritjoe said…**_
> Some clear policy with regard to how we subscribe and unsubscribe addresses to the FOSSRIT Fedora mailing list in particular.

I think it's best if we don't manually subscribe people to this list unless requested. Since it's a discussion list, it might be more specific to our community and not relevant for wider outreach work. We could try to better document this, though (e.g. "how to join the conversation").

> _**@ritjoe said…**_
> Some clear understanding as to how the list is to be used.

I think promoting it as a discussion / community list is best. We don't have a public announcement list other than what @schneidy collects on his private list. So, announcements can still go to this list, but it's more "internal" than "external".

I think the solution is being explicit in its promotion as a discussion list, not only for announcements.

**Comment by jwflory** [comment]
This exists in the IRC channel topic and is also linked on the existing http://foss.rit.edu website. I'm closing this issue as complete. :clapper:
